test/integration/docker/dsd_listening.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+import docker
+import os
+
+# The DOCKER_IMAGE envvar is needed to specify what
+# image to test
+
+SOCKET_PATH = "/tmp/statsd.socket"
+
+COMMON_ENVIRONMENT = [
+    "DD_DD_URL=http://dummy",
+    "DD_API_KEY=dummy",
+]
+
+ENVIRONMENTS = {
+    "udp": [],
+    "uds": [
+        "DD_DOGSTATSD_SOCKET=" + SOCKET_PATH,
+        "DD_DOGSTATSD_PORT=0"
+    ],
+    "both": [
+        "DD_DOGSTATSD_SOCKET=" + SOCKET_PATH,
+        "DD_DOGSTATSD_PORT=8125"
+    ]
+}
+
+containers = {}
+client = {}
+
+
+def setUpModule():
+    global containers
+    global client
+
+    client = docker.from_env()
+
+    for name, env in ENVIRONMENTS.iteritems():
+        containers[name] = client.containers.run(
+            os.environ.get('DOCKER_IMAGE'),
+            detach=True,
+            environment=COMMON_ENVIRONMENT + env,
+            auto_remove=True
+        )
+
+
+def tearDownModule():
+    global containers
+    global client
+
+    for _, container in containers.iteritems():
+        container.stop()
+
+
+def waitUntilListening(container, retries=20):
+    for x in range(0, retries):
+        out = container.exec_run(cmd="netstat -a")
```

Comment:
Why using netstat instead of lsof ?

Reply:
netstat is already shipped in alpine, while lsof had to be installed. this shaves off the install time from the test time

Reply:
Smart.